extension/lib/doc.js

```diff
@@ -0,0 +1,119 @@
+            throw `Invalid ${containerId}, should be [0, ${this.readableDomIds.length})`;
+        }
+        return $("#" + this.readableDomIds[containerId]);
+    }
+    
+    /*
+    Returns: Total words in document (int)
+    Calculates total words in the document
+    */
+    calcTotalWords(readableDomIds) {
+        var dict = {};
+        var total_words = 0;
+        var stop_words = "succeeded a about above after again against all am an and any are aren't as at be because been before being below between both but by can't cannot could couldn't did didn't do does doesn't doing don't down during each few for from further had hadn't has hasn't have haven't having he he'd he'll he's her here here's hers herself him himself his how however how's i i'd i'll i'm i've if in into is isn't it it's its itself let's me more most mustn't my myself no nor not of off on once only or other ought our ours ourselves out over own same shan’t she she'd she'll she's should shouldn't so some such than that that's the their theirs them themselves then there there's these they they'd they'll they're they've this those through to too under until up very was wasn't we we'd we'll we're we've were weren't what what's when when's where where's which while who who's whom why why's with won't would wouldn't you you'd you'll you're you've your yours yourself yourselves";
+        for (var section in readableDomIds) {
+            let text = this.getContainer(section).text();
+            let wordRegex = /\b\w+\b/g;
+            let wordList = text.match(wordRegex);
+            for (var i in wordList) {
+                if (stop_words.includes(wordList[i].toLowerCase())) {
+                    // Don't include stop words
+                }
+                else if (dict[wordList[i]]) { // Word is in dictionary
+                    dict[wordList[i]]++;
+                } else {
+                    dict[wordList[i]] = 1;
+                }
+            }
+            total_words += wordList.length;
+        }
+        this.dict = dict; // Set class attribute "dict"
+        return total_words // Set total words for use elsewhere (like Display)
+    }
+
+    /*
+    Returns: string[], given 
+    Determines the keywords of the document (using simple term frequency filter)
+    */
+    setKeyWords(dict) { 
+        // 1. Determine keywords
+        let keywords = [];
+        let keywordFilter = 0.005; // Any word below this frequency is a keyword
+        for (var word in dict) {
+            dict[word] /= this.total_words; // Divide frequency by total words
+            // console.log(dict[word]);
+            if (dict[word] < keywordFilter) {
+                keywords.push(word.toLowerCase());
+            };
+        };
+        return keywords;
+    };
+
+    /*
+    Params: Current container, start and end of tracker (jQuery element, int, int)
+    Highlights the keywords within the tracked sentence. 
+    */
+    highlightKeyWords(container, start, end) {
```

Comment:
this method i prefer to be in content_script.js
initially i was expecting doc.js to be "all the NLP logic. no UI prettiness"
but now we also add the UI drawing stuff in this class.
another downside is when someone wants to do ui stuff they'll collide w/ someone who's trying to do more nlp wizardry.

Can leave a TODO for future work, but might be something like this:
- doc.js identifies the keyword offsets within the container and return them as a list
- content_script.js does the ui highlighting.

What do you think?

Reply:
Yep I agree with internal logic 100%, was mostly a time consideration. Actually ended up not being hard, will include in next pr

Reply:
I actually was able to just move the entire func into contentScript, which just gets the keywords from the doc class. Does that jive with what you were thinking?

Reply:
yeah!
i was thinking a slightly different version where u return the _indices_ of the interesting keywords. when later on u want to use fancier nlp-based logic u are completely shielding the ui-interested people from the nlp logic.
but im fine w/ this rn.

Reply:
Oo yes I like that for later as well